bundles/org.eclipse.equinox.common/src/org/eclipse/core/internal/boot/PlatformURLConnection.java

```diff
@@ -164,39 +163,31 @@ private void copyToCache() throws IOException {
 		} else
 			tgt = cachedURL.getFile();
 		File tgtFile = null;
-		FileOutputStream tgtos = null;
 
 		boolean error = false;
 		long total = 0;
 
-		try {
+		try (InputStream srcis = src.openStream()) {
 			if (DEBUG && DEBUG_CACHE_COPY) {
 				if (isJar)
 					debug("Caching jar as " + tgt); //$NON-NLS-1$
 				else
 					debug("Caching as " + tgt); //$NON-NLS-1$
 			}
-
-			srcis = src.openStream();
 			byte[] buf = new byte[BUF_SIZE];
 			int count = srcis.read(buf);
 
 			tgtFile = new File(tgt);
-			tgtos = new FileOutputStream(tgtFile);
-
-			while (count != -1) {
-				total += count;
-				tgtos.write(buf, 0, count);
-				count = srcis.read(buf);
+			try (FileOutputStream tgtos = new FileOutputStream(tgtFile)){
+				while (count != -1) {
```

Comment:
Can't we use Files.copy() here?
It accepts different combinations of streams and Paths.